### PR TITLE
hcl: Fixed issue with empty arrays and modules

### PR DESCRIPTION
**Pull request by xescugc**
It was always trying to access the first element and not checking if it was empty before. Now it'll check and if it's empty it'll directly use a variable for it

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## [Unreleased]
 
+### Fixed
+
+- Empty array values on modules now are generated correctly and not failing
+  ([PR #147](https://github.com/cycloidio/terracognita/pull/147))
```
Comment (tormath1) [inline CHANGELOG.md:6]:
```suggestion
  ([PR #174](https://github.com/cycloidio/terracognita/pull/174))
```

**Review by kerak19 — APPROVED** [review]

File: hcl/writer.go [hunk 2]
```diff
@@ -286,6 +288,16 @@ func walkVariables(cfg map[string]interface{}, validVariables map[string]struct{
 		case map[string]interface{}:
 			cfg[key] = walkVariables(v, validVariables, currentKey, variables)
 		case []interface{}:
+			if len(v) == 0 {
+				if hasKey(validVariables, currentKey) {
+					varName := strings.ReplaceAll(currentKey, ".", "_")
+					variables[varName] = map[string]interface{}{
+						"default": cfg[key],
+					}
+					cfg[key] = fmt.Sprintf("${var.%s}", varName)
```
Comment (tormath1) [inline hcl/writer.go:297]: is it mandatory to keep this old notation `${...}` ?

Comment (xescugc) [reply hcl/writer.go:297]: Yes, because then the formatter removes it, so from `"key": "${var.name}"` to `"key": var.name`.

**Comment by xescugc** [comment]
Done

**Comment by tormath1** [comment]
@xescugc ok to rebase :)

**Comment by xescugc** [comment]
Done

**Review by tormath1 — APPROVED** [review]

**Pull request merged**